include/zephyr/bluetooth/l2cap.h

```diff
@@ -545,6 +591,34 @@ int bt_l2cap_chan_disconnect(struct bt_l2cap_chan *chan);
  */
 int bt_l2cap_chan_send(struct bt_l2cap_chan *chan, struct net_buf *buf);
 
+/** @brief Give credits to the remote
+ *
+ *  Only available for channels using @ref bt_l2cap_chan_ops.seg_recv.
+ *
+ *  Each credit given allows the peer to send one segment.
+ *
+ *  This function depends on a valid @p chan object. Make sure to
+ *  default-initialize or memset @p chan when allocating or reusing it
+ *  for new connections.
+ *
+ *  Credits can be given before calling @ref bt_l2cap_chan_connect, @ref
+ *  bt_l2cap_ecred_chan_connect or returning from @ref
+ *  bt_l2cap_server.accept. Doing so will adjust the 'initial credits'
+ *  sent in the connection PDU.
+ *
+ *  Must not be called after calling @ref bt_l2cap_chan_connect, @ref
+ *  bt_l2cap_ecred_chan_connect or
+ *  returning @ref bt_l2cap_server.accept until @ref
+ *  bt_l2cap_chan_ops.connected event.
+ *
+ *  Adding zero credits is not allowed.
+ *
+ *  ASYNC MAYBLOCK
```

Comment:
```
 * @note This function is asynchronous but may block in certain conditions.
```
I'm wondering what's the use of this statement anyway, async but may block kinda means "async but not really".